test/parallel/test-http-response-multiheaders.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const common = require('../common');
+const http = require('http');
+const assert = require('assert');
+
+// Test that certain response header fields do not repeat
+const norepeat = [
+  'retry-after',
+  'etag',
+  'last-modified',
+  'server',
+  'age',
+  'expires'
+];
+
+const server = http.createServer(function(req, res) {
+  for (let name of norepeat) {
+    res.setHeader(name, ['A', 'B']);
+  }
```

Comment:
Could be a little more rigorous by repeating the loop.  You should perhaps also test `res.writeHead()` and combinations thereof.